src/MongoDB.Driver.Legacy/MongoServerSettings.cs

```diff
@@ -139,6 +139,11 @@ public bool AllowInsecureTls
             set
             {
                 if (_isFrozen) { throw new InvalidOperationException("MongoServerSettings is frozen."); }
+                if (value && SslSettings != null)
+                {
+                    // Otherwise, the user will have to manually set CheckCertificateRevocation to false
+                    SslSettings.CheckCertificateRevocation = false;
+                }
```

Comment:
The existence of this side effect is somewhat disturbing as a caller of the setter of this property may not know that there is an invisible side effect taking place.

Reply:
We usually do not use the properties inside the class and use the backing field directly. You could code this as:

    if (value)
    {
        _sslSettings = _sslSettings ?? new SslSettings();
        _sslSettings.CheckCertificateRevocation = false;
    }

Reply:
I've also added a comment to clarify this side effect. Resolving!